lib/wallets/wallet/impl/solana_wallet.dart

```diff
@@ -0,0 +1,362 @@
+          SystemInstruction.transfer(fundingAccount: keyPair.publicKey, recipientAccount: recipientPubKey, lamports: txData.amount!.raw.toInt()),
+          ComputeBudgetInstruction.setComputeUnitPrice(microLamports: txData.fee!.raw.toInt()),
+        ],
+      );
+
+      final txid = await rpcClient.signAndSendTransaction(message, [keyPair]);
+      return txData.copyWith(
+        txid: txid,
+      );
+    } catch (e, s) {
+      Logging.instance.log(
+        "$runtimeType Solana confirmSend failed: $e\n$s",
+        level: LogLevel.Error,
+      );
+      rethrow;
+    }
+  }
+
+  @override
+  Future<Amount> estimateFeeFor(Amount amount, int feeRate) async {
+    if (info.cachedBalance.spendable.raw == BigInt.zero) {
+      return Amount(
+        rawValue: BigInt.zero,
+        fractionDigits: cryptoCurrency.fractionDigits,
+      );
+    }
+
+    final rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+    final fee = await rpcClient.getFees();
+
+    return Amount(
+      rawValue: BigInt.from(fee.value.feeCalculator.lamportsPerSignature),
+      fractionDigits: cryptoCurrency.fractionDigits,
+    );
+  }
+
+  @override
+  Future<FeeObject> get fees async {
+    final rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+    final fees = await rpcClient.getFees();
+    return FeeObject(
+        numberOfBlocksFast: 1,
+        numberOfBlocksAverage: 1,
+        numberOfBlocksSlow: 1,
+        fast: fees.value.feeCalculator.lamportsPerSignature,
+        medium: fees.value.feeCalculator.lamportsPerSignature,
+        slow: fees.value.feeCalculator.lamportsPerSignature
+    );
+  }
+
+  @override
+  Future<bool> pingCheck() {
+    return Future.value(false);
+  }
+
+  @override
+  FilterOperation? get receivingAddressFilterOperation =>
+      FilterGroup.and(standardReceivingAddressFilters);
+
+  @override
+  Future<void> recover({required bool isRescan}) async {
+    await refreshMutex.protect(() async {
+      var addressStruct = await _getCurrentAddress();
+
+      await mainDB.updateOrPutAddresses([addressStruct]);
+
+      if (info.cachedReceivingAddress != addressStruct.value) {
+        await info.updateReceivingAddress(
+          newAddress: addressStruct.value,
+          isar: mainDB.isar,
+        );
+      }
+
+      await Future.wait([
+        updateBalance(),
+        updateChainHeight(),
+        updateTransactions(),
+      ]);
+    });
+  }
+
+  @override
+  Future<void> updateBalance() async {
+    try {
+      var rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+      var balance = await rpcClient.getBalance(info.cachedReceivingAddress);
+
+      // Rent exemption of Solana
+      final accInfo = await rpcClient.getAccountInfo((await _getKeyPair()).address);
+      final minimumRent = await rpcClient.getMinimumBalanceForRentExemption(accInfo.value!.data.toString().length);
+      var spendableBalance = balance.value - minimumRent;
```

Comment:
What if we want to send more than the minimum rent out of the wallet?

Reply:
The account have to be closed to fully send all funds.

Reply:
Can we actually do that?